sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -570,82 +579,94 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
   }
 
   /** Load built-in test tables into the SparkSession. */
-  private def loadTestData(session: SparkSession): Unit = {
+  private def loadTestData(session: SparkSession, testTables: Seq[String]): Unit = {
     import session.implicits._
 
-    (1 to 100).map(i => (i, i.toString)).toDF("key", "value").createOrReplaceTempView("testdata")
-
-    ((Seq(1, 2, 3), Seq(Seq(1, 2, 3))) :: (Seq(2, 3, 4), Seq(Seq(2, 3, 4))) :: Nil)
-      .toDF("arraycol", "nestedarraycol")
-      .createOrReplaceTempView("arraydata")
-
-    (Tuple1(Map(1 -> "a1", 2 -> "b1", 3 -> "c1", 4 -> "d1", 5 -> "e1")) ::
-      Tuple1(Map(1 -> "a2", 2 -> "b2", 3 -> "c2", 4 -> "d2")) ::
-      Tuple1(Map(1 -> "a3", 2 -> "b3", 3 -> "c3")) ::
-      Tuple1(Map(1 -> "a4", 2 -> "b4")) ::
-      Tuple1(Map(1 -> "a5")) :: Nil)
-      .toDF("mapcol")
-      .createOrReplaceTempView("mapdata")
-
-    session
-      .read
-      .format("csv")
-      .options(Map("delimiter" -> "\t", "header" -> "false"))
-      .schema("a int, b float")
-      .load(testFile("test-data/postgresql/agg.data"))
-      .createOrReplaceTempView("aggtest")
-
-    session
-      .read
-      .format("csv")
-      .options(Map("delimiter" -> "\t", "header" -> "false"))
-      .schema(
-        """
-          |unique1 int,
-          |unique2 int,
-          |two int,
-          |four int,
-          |ten int,
-          |twenty int,
-          |hundred int,
-          |thousand int,
-          |twothousand int,
-          |fivethous int,
-          |tenthous int,
-          |odd int,
-          |even int,
-          |stringu1 string,
-          |stringu2 string,
-          |string4 string
-        """.stripMargin)
-      .load(testFile("test-data/postgresql/onek.data"))
-      .createOrReplaceTempView("onek")
-
-    session
-      .read
-      .format("csv")
-      .options(Map("delimiter" -> "\t", "header" -> "false"))
-      .schema(
-        """
-          |unique1 int,
-          |unique2 int,
-          |two int,
-          |four int,
-          |ten int,
-          |twenty int,
-          |hundred int,
-          |thousand int,
-          |twothousand int,
-          |fivethous int,
-          |tenthous int,
-          |odd int,
-          |even int,
-          |stringu1 string,
-          |stringu2 string,
-          |string4 string
-        """.stripMargin)
-      .load(testFile("test-data/postgresql/tenk.data"))
-      .createOrReplaceTempView("tenk1")
+    if (testTables.contains("testdata")) {
+      (1 to 100).map(i => (i, i.toString)).toDF("key", "value").createOrReplaceTempView("testdata")
+    }
+
+    if (testTables.contains("arraydata")) {
+      ((Seq(1, 2, 3), Seq(Seq(1, 2, 3))) :: (Seq(2, 3, 4), Seq(Seq(2, 3, 4))) :: Nil)
+        .toDF("arraycol", "nestedarraycol")
+        .createOrReplaceTempView("arraydata")
```

Comment:
To avoid the overhead of per-session init, we cannot just move these local temp views into a session-independent place, e.g., global temp views?

Reply:
The key point of conflict is not this. For example, test case A will create view testdata, and test case B will also create view testdata. However, the schema information of the two testdata is different. If the same session is shared globally, it will cause conflicts, especially in parallel execution.

Reply:
> If the same session is shared globally

I think that's not what @maropu means. We still create a fresh session for each testing file, but the testing views are created as global temp view, which are shared between all sessions.

Reply:
@cloud-fan Thanks. I under @maropu 's suggestion now. I will try to use `createGlobalTempView` and shared these views between all sessions.

Reply:
Sorry, but I missed your reply, @beliefer. Yea, that's what I wanted to say, thanks, @cloud-fan . I'll check later.

Reply:
@cloud-fan @maropu 
I tried use `createGlobalTempView` but failed.
If we want share the global temp view, we must not use `sparkSession.newSession()`.
Because every test case use a new session by `sparkSession.newSession()`.
If all the test case use the same `sparkSession`, another issue will occur. such as:
test case A create view testdata, and test case B will also create view testdata. However, the schema information of the two testdata is different. If the same session is shared globally, it will cause conflicts, especially in parallel execution.

Reply:
> we must not use sparkSession.newSession().

Why? All the `SparkSession`s share one `SharedState`, and the global temp views are stored in `SharedState`.

Reply:
Because some test case create a table or view who has the same name as the global temp view. The schema is different.

Reply:
@cloud-fan 
When use `sparkSession.newSession()`, will failed. such as:
```
23:25:05.078 ERROR org.apache.spark.sql.SQLQueryTestSuite: Error using configs: 
[info] - operators.sql *** FAILED *** (5 seconds, 722 milliseconds)
[info]   operators.sql
[info]   Expected "struct<[(- key):int,(+ key):int]>", but got "struct<[]>" Schema did not match for query #4
[info]   select -key, +key from testdata where key = 2: -- !query
[info]   select -key, +key from testdata where key = 2
[info]   -- !query schema
[info]   struct<>
[info]   -- !query output
[info]   org.apache.spark.sql.AnalysisException
[info]   Table or view not found: testdata; line 1 pos 23 (SQLQueryTestSuite.scala:464)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:530)
```

Reply:
```
val a = spark.sql("show views;")
a.show();
```
+---------+---------+-----------+
|namespace| viewName|isTemporary|
+---------+---------+-----------+
|         |  aggtest|       true|
|         |arraydata|       true|
|         |  mapdata|       true|
|         |     onek|       true|
|         |    tenk1|       true|
+---------+---------+-----------+

```
val a2 = localSparkSession.sql("show views;")
a2.show();
```
+---------+--------+-----------+
|namespace|viewName|isTemporary|
+---------+--------+-----------+
+---------+--------+-----------+
Maybe I lost some thing?

Reply:
like this? https://github.com/apache/spark/compare/master...maropu:SPARK-31291

Reply:
> master...maropu:SPARK-31291

This is a method but will cause too many changes. After a discussion offline between @cloud-fan and me, I will try to use `df.write.saveAsTable` replace `df.createTempView`.

Reply:
@cloud-fan @maropu I have used `df.write.saveAsTable` replace `df.createTempView`.
Because the origin temp view changed to tables, I have to regenerate some golden files.